codegen/src/api/mod.rs

```diff
@@ -78,18 +87,28 @@ where
     generator.generate_runtime(item_mod, derives)
 }
 
+/// Create the API for interacting with a Substrate runtime.
+///
+/// **Note:** Prefer using [generate_runtime_api] for static metadata use-cases.
 pub struct RuntimeGenerator {
     metadata: RuntimeMetadataV14,
 }
 
 impl RuntimeGenerator {
+    /// Create a new runtime generator from the provided metadata.
```

Comment:
Perhaps a note here instead of above like "If you have a path to metadata, you can use `generate_runtime_api` to load and use it"?